### PR TITLE
12696 Delete all test modules after integration tests

### PR DESCRIPTION
**Pull request by avstudnitz**
### Description
Several test modules are created on the fly during startup of every
integration test run, but they are never deleted. This PR
changes this, modules are deleted when the process ends.

Things we thought about but didn't implement:

- Putting the modules into `dev/tests/integration/tmp/*/app/code/`: we'd need to modify the autoloader and the component registrar which might have caused side effects. Plus, the approach we tried didn't work out.
- Leave the modules in `dev/tests/integration/_files` and register them from there: same as above.
- Remove the modules from a `dev/tests/integration/framework/removeTestModules.php`: PhpUnit doesn't offer an option to run a script after all tests have been executed.

### Fixed Issues
1. magento/magento2#12696: Integration tests create stub modules in app/code

### Manual testing scenarios
1. Run any integration test
2. Verify that during the test run, some directories names `TestModule*`, i.e. `TestModuleSample`, exist in `app/code/Magento` (they are created automatically when the tests are started).
3. Verify that those directories are gone after the tests have ended.

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by avstudnitz** [comment]
I added a check, so the modules are NOT deleted if the integration tests are run parallelly.

File: dev/tests/integration/framework/deployTestModules.php [hunk 1]
```diff
@@ -38,3 +39,28 @@
...
+    $iterator = new DirectoryIterator($pathToCommittedTestModules);
+    /** @var SplFileInfo $file */
+    foreach ($iterator as $file) {
+        if ($file->isDir() && !in_array($file->getFilename(), ['.', '..'])) {
+            $targetDirPath = $pathToInstalledMagentoInstanceModules . '/' . $file->getFilename();
+            $filesystem->remove($targetDirPath);
+        }
+    }
+    unset($iterator, $file);
+}
```
Comment (mhauri) [inline dev/tests/integration/framework/deployTestModules.php]: Expected 1 newline at end of file; 0 found

**Review by mhauri — CHANGES_REQUESTED** [review]
Hi @avstudnitz thank you for your contribution, can you please fix the issues from the static tests?

FILE: deployTestModules.php

Line: 45 | Line exceeds maximum limit of 120 characters; contains 121 characters
Line:  66 | Expected 1 newline at end of file; 0 found

**Comment by avstudnitz** [comment]
@mhauri Resolved. Thanks @sidolov for doing the changes.

**Review by mhauri — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @mhauri, thank you for the review.
[ENGCOM-3132](https://jira.corp.magento.com/browse/ENGCOM-3132) has been created to process this Pull Request

**Comment by danslo** [comment]
Did this ever manage to get merged? I can't see internal ENGCOM-3132.

**Comment by okorshenko** [comment]
Hi @danslo 
We did 13 attempts to deliver this change, but unfortunately, these changes can't pass our internal automated tests. Our team is working on improvements to this solution to align it with our internal testing infrastructure. Latest progress update on this PR was 13 hours ago

**Comment by danslo** [comment]
Thanks for the update!

**Review by sivaschenko — CHANGES_REQUESTED** [review]
There is a fatal error duirng integration tests run:

    [Integration-10-ce] PHP Fatal error:  Uncaught Magento\Framework\Exception\FileSystemException: The "/var/www/html/app/code/Magento/TestModuleWysiwygConfig/etc/adminhtml/di.xml" file doesn't exist. in /var/www/html/lib/internal/Magento/Framework/Filesystem/File/Read.php:76
    [Integration-10-ce] Stack trace:
    [Integration-10-ce] #0 /var/www/html/lib/internal/Magento/Framework/Filesystem/File/Read.php(62): Magento\Framework\Filesystem\File\Read->assertValid()
    [Integration-10-ce] #1 /var/www/html/lib/internal/Magento/Framework/Filesystem/File/Read.php(52): Magento\Framework\Filesystem\File\Read->open()
    [Integration-10-ce] #2 /var/www/html/lib/internal/Magento/Framework/Filesystem/File/ReadFactory.php(44): Magento\Framework\Filesystem\File\Read->__construct('/var/www/html/a...', Object(Magento\Framework\Filesystem\Driver\File))
    [Integration-10-ce] #3 /var/www/html/lib/internal/Magento/Framework/Config/FileIterator.php(69): Magento\Framework\Filesystem\File\ReadFactory->create('/var/www/html/a...', 'file')
    [Integration-10-ce] #4 /var/www/html/lib/internal/Magento/Framework/Config/Reader/Filesystem.php(147): Magento\Framework\Config\FileIterator->current()
    [Integration-10-ce] #5 /var/www/html in /var/www/html/lib/internal/Magento/Framework/Filesystem/File/Read.php on line 76
    [Integration-0-b2b] b2b Integration Integration 0-1 exit code: 0

@nmalevanec could you please help to resolve this problem

**Pull request merged**